homeassistant/components/fan/alfawise.py

```diff
@@ -0,0 +1,121 @@
+"""
+Alfawise fan platform (fan is used as a humidifier).
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/fan.alfawise/
+"""
+import logging
+import ipaddress
+from typing import Any
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.fan import (SPEED_LOW, SPEED_HIGH,
+                                          FanEntity, SUPPORT_SET_SPEED)
+from homeassistant.const import CONF_DEVICES, CONF_NAME, CONF_MAC
+from homeassistant.const import STATE_OFF
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.exceptions import PlatformNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyAlfawise==0.5-beta']
+
+
+def isip(value: Any) -> str:
```

Comment:
Checks like this should be part of `config_validation.py` because it could be re-used by other platforms.

Reply:
This is what I asked to @MartinHjelmare in the previous PR #13923  . He tells me to put it in the platform file instead of the config_validation.py file

Reply:
The problem is that if it part of the platform then the validator is not documented and the chance is high that somebody is doing the same again. Also, it's not tested.

Reply:
We usually wait with adding a new validator to the helper module until at least two other modules require the validator.

But it's not a big deal.